pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotRealtimeSegmentManager.java

```diff
@@ -138,15 +138,22 @@ private synchronized void assignRealtimeSegmentsToServerInstancesIfNecessary()
         for (String instanceId : instancesInResource) {
           InstanceZKMetadata instanceZKMetadata = _pinotHelixResourceManager.getInstanceZKMetadata(instanceId);
           String groupId = instanceZKMetadata.getGroupId(resource);
+          String partitionId = instanceZKMetadata.getPartition(resource);
+          AbstractTableConfig realtimeTableConfig =
+              ZKMetadataProvider.getRealtimeTableConfig(_pinotHelixResourceManager.getPropertyStore(), resource);
+          if (instancesInResource.size() == Integer.parseInt(realtimeTableConfig.getValidationConfig().getReplication())) {
```

Comment:
Well just pull whatever is in the instance config. It's not guaranteed to always be numerical. :)